doc/source/serve/architecture.md

```diff
@@ -106,13 +106,13 @@ servers.  You can use your own load balancer on top of Ray Serve.
 This architecture ensures horizontal scalability for Serve. You can scale your HTTP ingress by adding more nodes and scale your model inference by increasing the number
 of replicas via the `num_replicas` option of your deployment.
 
-### How do ServeHandles work?
+### How do deployment handles work?
 
-{mod}`ServeHandles <ray.serve.handle.RayServeHandle>` wrap a handle to a "router" on the
-same node which routes requests to replicas for a deployment. When a
-request is sent from one replica to another via the handle, the
+The {mod}`DeploymentHandle <ray.serve.handle.DeploymentHandle>` object wraps a handle to a
+"router" that implements client-side load balancing of requests to replicas for a deployment.
+When a request is sent from one replica to another via the handle, the
 requests go through the same data path as incoming HTTP requests. This enables
-the same deployment selection and batching procedures to happen. ServeHandles are
+the same replica selection and batching procedures to happen. Deployment handles are
 often used to implement [model composition](serve-model-composition).
```

Comment:
This line sounds a bit weird to me: "Deployment handles are often used to implement model composition". Are we trying to emphasize how when there's multiple deployments composed together, you use deployment handles to send requests between them?

Reply:
deleted this, it's redundant with section above.